lib/transloadit/assembly.rb

```diff
@@ -43,22 +45,25 @@ def steps
   #     {template}[https://transloadit.com/templates] to use instead of
   #     specifying params here directly
   #
-  def create!(*ios)
-    params = _extract_options!(ios)
+  def create!(*ios, **params)
     params[:steps] = _wrap_steps_in_hash(params[:steps]) if !params[:steps].nil?
 
     extra_params = {}
-    # update the payload with file entries
-    ios.each_with_index {|f, i| extra_params.update :"file_#{i}" => f }
-
     extra_params.merge!(self.options[:fields]) if self.options[:fields]
 
-    _do_request(
-      '/assemblies',
-      params,
-      'post',
-      extra_params
-    ).extend!(Transloadit::Response::Assembly)
+    trials = 1 + (self.options[:rate_limit_retries] || DEFAULT_RATE_LIMIT_RETRIES)
+    (1..trials).each do |trial|
```

Comment:
I was thinking to avoid having a user pass 0 for tries and then it doesn't run at all(which most likely wouldn't be intended). Having it behave that way might also push us to make it default to 1 when a user passes 0, and then in that case "intention revealing" might be lost.

However, for intention revealing I'd say yes it does sound like a good idea 👍